tests/cache/tests.py

```diff
@@ -2578,6 +2578,42 @@ def test_middleware(self):
         self.assertIsNotNone(result)
         self.assertEqual(result.content, b"Hello World 1")
 
+    def test_middleware_group(self):
+        def get_group2(*args, **kwargs):
+            print(args)
+            print(kwargs)
+            return "group2"
+
+        middleware_group1 = CacheMiddleware(hello_world_view, key_group="group1")
+        middleware_group2_callable = CacheMiddleware(hello_world_view, key_group=get_group2)
+
+        request = self.factory.get("/view/")
+
+        # Put the request through the request middleware
+        result = middleware_group1.process_request(request)
+        self.assertIsNone(result)  # Is None because there's no cache
+
+        response = hello_world_view(request, "1")
+
+        # Now put the response through the response middleware
+        response = middleware_group1.process_response(request, response)
+
+        # Repeating the request should result in a cache hit
+        result = middleware_group1.process_request(request)
+        self.assertIsNotNone(result)
+
+        # Put the request through the request middleware
+        result = middleware_group2_callable.process_request(request)
+        self.assertIsNone(result)  # Is None because there's no cache
+
+        # Now put the response through the response middleware
+        response = middleware_group2_callable.process_response(request, response)
+
+        # Repeating the request should result in a cache hit
+        result = middleware_group2_callable.process_request(request)
+        self.assertIsNotNone(result)
+
+
```

Comment:
We might have another test with for the utils to check what gets put on the cache.